# 1.) LOAD DATA, CREATE SEURAT OBJECT, UPDATE METADATA
## Download dataset from BioStudies database
The dataset is available from EMBL-EBI's BioStudies database (http://www.ebi.ac.uk/biostudies) under accession number E-MTAB-15970 (https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-15970).

Files '1-10_P4_barcodes.tsv.gz', '1-10_P4_features.tsv.gz', '1-10_P4_matrix.mtx.gz' were downloaded, renamed to 'barcodes.tsv.gz', 'features.tsv.gz', 'matrix.mtx.gz', respectively and transfered into a folder named "..\filtered_bc_matrix\"

### libraries
    library(dplyr)
    library(Seurat)
    library(patchwork)
    library(ggplot2)
    library(limma)
    library(ggrepel)

### set working directory (or create 'path' for analysis data storage)
create folders on the system, e.g. on a windows machine C:\user\[USERNAME]/Documents/R/R_Workspace/scRNAseq_Mouse_P4_wt/

    path <- "C:/Users/[USERNAME]/Documents/scRNAseq_Mouse_P4_wt/"

I prefer working with working directory, this works with R with default installation on a windows machine:

    setwd("~R/R_Workspace")

### load dataset
filtered cell ranger output UMI-count matrix

    Scn2aP4.data <- Read10X(data.dir = [YOUR FOLDER OF INTEREST, e.g. "C:/Users/[YOUR_USERNAME]/Downloads/filtered_bc_matrix/"]
    

### initialize Seurat objects
filter genes (features) expressed in <3 cells and cells expressing < 200 genes

    Scn2a.P4 <- CreateSeuratObject(counts = Scn2aP4.data, project = "Scn2aP4", min.cells = 3, min.features = 200)    

### update metadata, add grouping information, etc.
#### add sample_no to metadata

    Scn2a.P4[["sample_no"]] <- sub('.*-', '', rownames(Scn2a.P4@meta.data))

#### add genotype to metadata
    Idents(object = Scn2a.P4) <- "sample_no"

    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '1' = 'hom')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '2' = 'wt')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '3' = 'het')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '4' = 'wt')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '5' = 'het')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '6' = 'wt')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '7' = 'het')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '8' = 'hom')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '9' = 'wt')
    Scn2a.P4 <- RenameIdents(object = Scn2a.P4, '10' = 'het')    

    Scn2a.P4[["genotype"]] <- Idents(Scn2a.P4)
    
### add percent of mitochondrial gene counts for QC to metadata
    Scn2a.P4[["percent.mt"]] <- PercentageFeatureSet(Scn2a.P4, pattern = "^mt-")

### save raw pre-QC dataset
uncomment to performe, destination path for saved file needs to be adjusted accordingly

    #saveRDS(Scn2a.P4, file = "~/R/Scn2a_P4_meta.rds")


