# QC

### libraries
    # Libryries already activated before
        #library(dplyr)
        #library(Seurat)
        #library(patchwork)
        #library(ggplot2)
        #library(limma)
        #library(ggrepel)

    # libraries for QC
        library(ggExtra)
        library(cowplot)
        library(Matrix)


## Extract raw dataset QC metrics (metadata) summary


### get metadata

    meta <- Scn2a.P4@meta.data


####  define empty dataframe for metadata sammary    
    metadata.summary = data.frame(NA_col = rep(NA, 7))


#####  add columns with UMI-counts ('nCount_RNA') for each sample   
    for (sample_no in levels(as.factor(meta$sample_no))) {
        sample.summary <- c(sample_no, unname(summary(meta$nCount_RNA [which(meta$sample_no == sample_no)])))
        metadata.summary[ , sample_no] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(sample_no, "_nUMI")
        }

#####  add columns with UMI-counts grouped by genotype and for all the total dataset 
    for (genotype in levels(meta$genotype)) {
        sample.summary <- c(genotype, unname(summary(meta$nCount_RNA [which(meta$genotype == genotype)])))
        metadata.summary[ , genotype] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(genotype, "_nUMI")
        } 

    metadata.summary[ , "all_nUMI"] <- c("all", unname(summary(meta$nCount_RNA)))

#####  remove column NA_col containing NA only
    metadata.summary <- metadata.summary[, !(names(metadata.summary) =="NA_col")]
    # OR use: metadata.summary <- metadata.summary[, -1]

#####  add row names
    rownames(metadata.summary) <- c("sample", names(summary(meta$nCount_RNA [which(meta$sample_no == sample_no)])))

#####  add columns with gene-counts ('nFeature_RNA') for each sample, genotype, and total dataset
    for (sample_no in levels(as.factor(meta$sample_no))) {
        sample.summary <- c(sample_no, unname(summary(meta$nFeature_RNA [which(meta$sample_no == sample_no)])))
        metadata.summary[ , ncol(metadata.summary)+1] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(sample_no, "_nGene")
        }
                  
    for (genotype in levels(meta$genotype)) {
        sample.summary <- c(genotype, unname(summary(meta$nFeature_RNA [which(meta$genotype == genotype)])))
        metadata.summary[ , genotype] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(genotype, "_nGene")
        }
                  
    metadata.summary[ , "all_nGene"] <- c("all", unname(summary(meta$nFeature_RNA)))


#####  add columns with percent mitochondrial gene counts for each sample, genotype, and total dataset
    for (sample_no in levels(as.factor(meta$sample_no))) {
        sample.summary <- c(sample_no, unname(summary(meta$percent.mt [which(meta$sample_no == sample_no)])))
        metadata.summary[ , ncol(metadata.summary)+1] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(sample_no, "_pc.mito")
        }    
            
    for (genotype in levels(meta$genotype)) {
        sample.summary <- c(genotype, unname(summary(meta$percent.mt [which(meta$genotype == genotype)])))
        metadata.summary[ , genotype] <- sample.summary
        colnames(metadata.summary)[ncol(metadata.summary)] <- paste0(genotype, "_pc_mt")
        }
            
    metadata.summary[ , "all_pc_mt"] <- c("all", unname(summary(meta$percent.mt)))

####  store pre-QC summary
    write.table(metadata.summary, file = paste(path, "summary_metadata_raw.csv", sep = ""), 
                row.names = TRUE, col.names = TRUE, sep = ";")                              

##  adjust QC cut off values
thanks to https://matthieuxmoreau.github.io/EarlyPallialNeurogenesis/html-Reports/Quality_Control.html

####  Set ggplot theme as classic 
    theme_set(theme_classic())


####  add "percent.rb" (percent ribosomal gene counts) to metadata
    MB_N5[["percent.rb"]] <- PercentageFeatureSet(MB_N5, pattern = "(^Rp[sl]|^Mrp)")            
    MB_N5[["percent.ctrb"]] <- PercentageFeatureSet(MB_N5, pattern = "^Rp[sl]")            
    MB_N5[["percent.mtrb"]] <- PercentageFeatureSet(MB_N5, pattern = "^Mrp")

    
###  plot pre QC
    
####  Violinplots of QC parameters
    Idents(Scn2a.P4) <- "sample_no"
    
    VlnPlot(Scn2a.P4, features = c("nCount_RNA","nFeature_RNA", "percent.mt", "percent.rb"), raster = FALSE, ncol = 2 )

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/QC_Figs/QC_1_preQC_violins.png" width="400" height="400"/>
</p>


####  Relation between nunber of UMI and number of Genes per cell
    pA <- ggplot(MB_N5@meta.data, aes(x = nCount_RNA, y = nFeature_RNA)) + geom_point() + geom_smooth(method="lm")
    pA <- ggMarginal(pA, type = "histogram", fill="lightgrey")
    
    pB <- ggplot(MB_N5@meta.data, aes(x=log10(nCount_RNA), y=log10(nFeature_RNA))) + geom_point() + geom_smooth(method="lm")
    pB <- ggMarginal(pB, type = "histogram", fill="lightgrey")
    
    plot_grid(plotlist = list(pA,pB), ncol=2, align='h', rel_widths = c(1, 1))

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/QC_Figs/QC_2_nGenes_nUMI_preQC_scatter.png" width="800" height="400"/>
</p>


###  low quality cell filtering
####  extract modifiable QC statistics for modification
    Cell.QC.Stat <- Scn2a.P4@meta.data


####  filtering cells by mitochondrial reads content [histogram-based threshold assignment]
    ##  set low and high threshold
        max.mito.thr <- median(Cell.QC.Stat$percent.mt) + 3*mad(Cell.QC.Stat$percent.mt)    # max.mito.thr <- 12.06116
        min.mito.thr = min(Cell.QC.Stat$percent.mt)                                         # min.mito.thr = 0
            
        p1 <- ggplot(Cell.QC.Stat, aes(x=nFeature_RNA, y=percent.mt)) +
              geom_point() +
              geom_hline(aes(yintercept = max.mito.thr), colour = "red", linetype = 2, linewidth = 1) +
              geom_hline(aes(yintercept = min.mito.thr), colour = "red", linetype = 2) +
              annotate(geom = "text", label = paste0(as.numeric(table(Cell.QC.Stat$percent.mt > max.mito.thr | Cell.QC.Stat$percent.mt < min.mito.thr)[2])," cells removed\n",
                                                     as.numeric(table(Cell.QC.Stat$percent.mt > max.mito.thr | Cell.QC.Stat$percent.mt < min.mito.thr)[1])," cells remain"), x = 4000, y = 4)
            
        ggMarginal(p1, type = "histogram", fill="lightgrey", bins=100)

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/QC_Figs/QC_3_mito_filt.png" width="400" height="400"/>
</p>

     ##  display upper threshold (cut-off) of % mithochondrial reads
         paste("max.mito.thr = ", round(max.mito.thr, digits = 3), "% mitochondrial gene reads")

[1] "max.mito.thr =  12.061 % mitochondrial gene reads"         

     ##  Filter cells based on mitochondrial content thresholds
         Cell.QC.Stat <- Cell.QC.Stat %>% filter(percent.mt < max.mito.thr) %>% filter(percent.mt >= min.mito.thr)       


####  filtering cells by number of genes (nFeature_RNA) and transcripts (nCount_RNA) [histogram-based threshold assignment]
    ##  Set low and high thresholds on the number of detected genes
        min.Genes.thr <- median(log10(Cell.QC.Stat$nFeature_RNA)) - 1.6*mad(log10(Cell.QC.Stat$nFeature_RNA))
        max.Genes.thr <- median(log10(Cell.QC.Stat$nFeature_RNA)) + 2.25*mad(log10(Cell.QC.Stat$nFeature_RNA))    
        
    ##  Set high threshold on the number of transcripts
        min.nUMI.thr <- median(log10(Cell.QC.Stat$nCount_RNA)) - 1.27*mad(log10(Cell.QC.Stat$nCount_RNA))
        max.nUMI.thr <- median(log10(Cell.QC.Stat$nCount_RNA)) + 2.85*mad(log10(Cell.QC.Stat$nCount_RNA))

    ##  display thresholds:
        paste("min.UMI.thr:", round(min.nUMI.thr, 3), "; max.UMI.thr:", round(max.nUMI.thr, 3), 
                      "; UMI per cell:", round(10^min.nUMI.thr,0), "-", round(10^max.nUMI.thr, 0))
                      
        paste("min.Genes.thr:", round(min.Genes.thr, 3), "; max.Genes.thr:", round(max.Genes.thr, 3), 
              "; Genes per cell:", round(10^min.Genes.thr,0), "-", round(10^max.Genes.thr, 0))

[1] "min.UMI.thr: 3.698 ; max.UMI.thr: 5.049 ; UMI per cell: 4994 - 111913"

[1] "min.Genes.thr: 3.3 ; max.Genes.thr: 4.016 ; Genes per cell: 1997 - 10372"

    ##  Gene/UMI scatter plot before filtering
        p2 <- ggplot(Cell.QC.Stat, aes(x=log10(nCount_RNA), y=log10(nFeature_RNA))) +
          geom_point() +
          geom_smooth(method="lm") +
          geom_hline(aes(yintercept = min.Genes.thr), colour = "red", linetype = 2, linewidth = 2) +
          geom_hline(aes(yintercept = max.Genes.thr), colour = "darkgreen", linetype = 2, linewidth = 2) +
          geom_vline(aes(xintercept = min.nUMI.thr), colour = "red", linetype = 2, linewidth = 2) +
          geom_vline(aes(xintercept = max.nUMI.thr), colour = "red", linetype = 2, linewidth = 2)
    
        ggMarginal(p2, type = "histogram", fill="lightgrey")



<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/QC_Figs/QC_4_feature_count_threshold_scatter.png" width="400" height="400"/>
</p>


####  filtering cells by deviation from linear model of (double log) nUMI/nGene relation
    ##  nUMI/nGene relationship
        lm.model <- lm(data = Cell.QC.Stat, formula = log10(nFeature_RNA) ~ log10(nCount_RNA))

    ##  plot nUMI/nGene and linear model        
        p3 <- ggplot(Cell.QC.Stat, aes(x=log10(nCount_RNA), y=log10(nFeature_RNA))) +
          geom_point() +
          geom_smooth(method="lm") +
          geom_hline(aes(yintercept = min.Genes.thr), colour = "green4", linetype = 2, linewidth = 1.5) +
          geom_hline(aes(yintercept = max.Genes.thr), colour = "green4", linetype = 2, linewidth = 1.5) +
          geom_vline(aes(xintercept = min.nUMI.thr), colour = "red", linetype = 2, linewidth = 1.5) +
          geom_vline(aes(xintercept = max.nUMI.thr), colour = "red", linetype = 2, linewidth = 1.5) +
          geom_abline(intercept = lm.model$coefficients[1] - 0.12 , slope = lm.model$coefficients[2], color="orange", linewidth = 2) +
          annotate(geom = "text", label = paste0(dim(Cell.QC.Stat)[1], " QC passed cells"), x = 4.0, y = 4.0)
        
        ggMarginal(p3, type = "histogram", fill="lightgrey")
        
    ##  cells to exclude
        Cell.QC.Stat$valideCells <- log10(Cell.QC.Stat$nFeature_RNA) > (log10(Cell.QC.Stat$nCount_RNA) * lm.model$coefficients[2] + (lm.model$coefficients[1] - 0.10))

    ##  display filtering plot
        p4 <- ggplot(Cell.QC.Stat, aes(x=log10(nCount_RNA), y=log10(nFeature_RNA))) +
              geom_point(aes(colour = valideCells)) +
              geom_smooth(method="lm") +
              geom_abline(intercept = lm.model$coefficients[1] - 0.10 , slope = lm.model$coefficients[2], color="orange", linewidth = 1.5) + 
              theme(legend.position="none") +
              annotate(geom = "text", label = paste0(as.numeric(table(Cell.QC.Stat$valideCells)[2]), " QC passed cells\n",
                                                     as.numeric(table(Cell.QC.Stat$valideCells)[1]), " QC filtered"), x = 4.0, y = 3.9)
            
        ggMarginal(p4, type = "histogram", fill="lightgrey")

<p align="center">
    <img src="https://github.com/mstockebrand/Df16A_Dopaminergic_scRNA-seq/blob/main/Analysis workflow/Figures/QC_Figs/QC_5_linear_model.png" width="400" height="400" hspace="10" >
    <img src="https://github.com/mstockebrand/Df16A_Dopaminergic_scRNA-seq/blob/main/Analysis workflow/Figures/QC_Figs/QC_6_linear_model_filtering.png" width="400" height="400" hspace="10" >
</p>


    ##  Remove unvalid cells
        Cell.QC.Stat <- Cell.QC.Stat %>% filter(valideCells)
            
    ##  Keep only valid cells in Seurat object
        MB_N5 <- subset(MB_N5, cells = rownames(Cell.QC.Stat))
            
    ##  Plot final QC metrics
        VlnPlot(MB_N5, features = c("nCount_RNA", "nFeature_RNA", "percent.mt", "percent.rb"), ncol = 2)
        
        p5 <- ggplot(MB_N5@meta.data, aes(x=log10(nCount_RNA), y=log10(nFeature_RNA))) + geom_point() + geom_smooth(method="lm")
        ggMarginal(p5, type = "histogram", fill="lightgrey")


<p align="center">
    <img src="https://github.com/mstockebrand/Df16A_Dopaminergic_scRNA-seq/blob/main/Analysis workflow/Figures/QC_Figs/QC_7_feature_UMI_scatter_QC.png" width="400" height="400" hspace="10" >
    <img src="https://github.com/mstockebrand/Df16A_Dopaminergic_scRNA-seq/blob/main/Analysis workflow/Figures/QC_Figs/QC_8_VlnPlot_QC.png" width="400" height="400" hspace="10" >
</p>
