#  Sample Integration and Normalization

high quality singlet dataset as input for Normalization and integration

###  libraries
    # Libryries already activated before
        #library(dplyr)
        #library(Seurat)
        #library(patchwork)
        #library(ggplot2)
        #library(limma)
        #library(ggrepel)

        # libraries for QC
            #library(ggExtra)
            #library(cowplot)
            #library(Matrix)

        # libraries for doublet removal
          library(DoubletFinder)
          library(glmGamPoi) 

###  Dataset integration and normalization
####  Split layers
    Scn2a.P4[["RNA"]] <- split(Scn2a.P4[["RNA"]], f = Scn2a.P4$sample_no)

###  Normalization with SCTransform
    Scn2a.P4 <- SCTransform(Scn2a.P4, method = "glmGamPoi",                            
                            vars.to.regress = c("nCount_RNA", "percent.mt"))

###  linear dimensionality reduction: PCA ----
    Scn2a.P4 <- RunPCA(Scn2a.P4)

####  Examine and visualize PCA results a few different ways
##### Check PC content
    print(Scn2a.P4[["pca"]], dims = 1:5, nfeatures = 10)

    ##PC_ 1 
    ##Positive:  Meg3, Cck, Nrgn, Stmn2, Crym, Basp1, Igfbpl1, Stmn1, Gap43, Tuba1a 
    ##Negative:  Fabp7, Apoe, Ptn, Dbi, Slc1a3, Ptprz1, Cst3, Aldoc, Atp1a2, Id3 
    ##PC_ 2 
    ##Positive:  Fabp7, Dbi, Ptprz1, Slc1a3, Apoe, Aldoc, Ptn, Cst3, Mt3, Slc1a2 
    ##Negative:  Sparc, Igfbp7, Col4a1, Gng11, Col4a2, Selenop, Itm2a, Pltp, Esam, Cldn5 
    ##PC_ 3 
    ##Positive:  Hmgb2, Top2a, Ube2c, Cenpf, Cks2, Mki67, Pclaf, Prc1, Cdk1, Cenpa 
    ##Negative:  Aldoc, Apoe, Meg3, Cst3, Ptn, Sparcl1, Sparc, Itm2a, Mt3, Slco1c1 
    ##PC_ 4 
    ##Positive:  Crym, Cck, Nrgn, Fabp7, Lpl, Zbtb20, Igfbpl1, Pou3f1, Atp2b1, Neurod6 
    ##Negative:  Reln, Maf, Sst, Npy, Meg3, Snhg11, Pnoc, Mef2c, Ndnf, Mia 
    ##PC_ 5 
    ##Positive:  Zbtb20, Mt1, Crym, Meig1, Ecrg4, Dynlrb2, Riiad1, Foxj1, 1700094D03Rik, Rsph1 
    ##Negative:  Meg3, Sst, Npy, Mef2c, Maf, Snhg11, Reln, Pnoc, Nxph1, Nrxn3 

##### Visualize Dim Loading (weight of genes in linear combinations within a PC)
    VizDimLoadings(Scn2a.P4, dims = 1:2, nfeatures = 50, reduction = "pca")

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/Integration_Figs/Int_1_PCA_dim_loadings.png" width="500" height="500"/>
</p>

##### Check dataset complexity caught in selected PCs, compare genotypes
    Idents(Scn2a.P4) <- "genotype"
    DimPlot(Scn2a.P4, reduction = "pca")
    DimPlot(Scn2a.P4, reduction = "pca", dims = c(49, 50))  

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/Integration_Figs/Int_2_PCA_DimPlots.png" width="800" height="383"/>
</p>

##### Check heterogeneity captures by first and last created PCs
    DimHeatmap(Scn2a.P4, dims = 1:9, cells = 500, balanced =TRUE)
    DimHeatmap(Scn2a.P4, dims = 42:50, cells = 500, balanced =TRUE)

<p align="center">
<img src="https://github.com/mstockebrand/GABRA5_scRNAseq_Mm_P4_hpc/blob/main/Analysis Workflow/Figures/Integration_Figs/Int_3_PCA_DimHeatmaps.png" width="800" height="383"/>
</p>    
